EU4toV2/Source/V2World/Decisions/HPMDecisions.cpp

```diff
@@ -0,0 +1,228 @@
+		Log(LogLevel::Debug) << "Could not open 02_NNM_gtfo.csv for writing!";
+	outLoc << ";;;;;;;;;;;;x,,,,,,,,,,,,,\n";
+	outLoc << "gtfo_on_title;Enable 'Release Vassal' decisions;;;;;;;;;;;;x,,\n";
+	outLoc << "gtfo_on_desc;The Release Vassal decisions allow you to release vassals, and are otherwise kept hidden from view to avoid having them clutter up your decision list.;;;;;;;;;;;;x\n";
+	outLoc << "gtfo_off_title;Disable 'Release Vassal' decisions;;;;;;;;;;;;x,,\n";
+	outLoc << "gtfo_off_desc;Your decision list will now be uncluttered!;;;;;;;;;;;;x,,\n";
+	outLoc << "gtfo_disabled_title;Deactivate 'Release Vassal' decisions;;;;;;;;;;;;x,,\n";
+	outLoc << "gtfo_disabled_desc;The 'Release Vassal' decisions will disappear from this game session and won't reappear again. You won't be able to use these decisions again for this game.;;;;;;;;;;;;x,,\n";
+
+	std::ofstream output("output/" + theConfiguration.getOutputName() + "/decisions/gtfo.txt");
+	if (!output.is_open())
+		throw std::runtime_error("Could not open gtfo.txt for writing!");
+	output << "political_decisions = {\n";
+	output << "\tgtfo_on = {\n";
+	output << "\t\tpicture = gtfo\n";
+	output << "\t\talert = no\n";
+	output << "\t\tpotential = {\n";
+	output << "\t\t\tOR = {\n";
+	output << "\t\t\t\thas_country_flag = government_decisions_on\n";
+	output << "\t\t\t\tai = yes\n";
+	output << "\t\t\t}\n";
+	output << "\t\t\tnum_of_vassals_no_substates = 1\n";
+	output << "\t\t\tis_vassal = no\n";
+	output << "\t\t\tNOT = { has_country_flag = gtfo has_country_flag = gtfo_disabled }\n";
+	output << "\t\t}\n";
+	output << "\t\tallow = { civilized = yes ai = no }\n";
+	output << "\t\teffect = { set_country_flag = gtfo }\n";
+	output << "\t}\n";
+	output << "\tgtfo_off = {\n";
+	output << "\t\tpicture = gtfo\n";
+	output << "\t\talert = no\n";
+	output << "\t\tpotential = {\n";
+	output << "\t\t\thas_country_flag = gtfo\n";
+	output << "\t\t\tNOT = { has_country_flag = gtfo_disabled }\n";
+	output << "\t\t}\n";
+	output << "\t\tallow = { ai = no }\n";
+	output << "\t\teffect = { clr_country_flag = gtfo }\n";
+	output << "\t}\n";
+	output << "\tgtfo_disabled = {\n";
+	output << "\t\tpicture = gtfo\n";
+	output << "\t\talert = no\n";
+	output << "\t\tpotential = {\n";
+	output << "\t\t\tOR = {\n";
+	output << "\t\t\t\thas_country_flag = government_decisions_on\n";
+	output << "\t\t\t\tai = yes\n";
+	output << "\t\t\t}\n";
+	output << "\t\t\thas_country_flag = gtfo\n";
+	output << "\t\t\tNOT = { has_country_flag = gtfo_disabled }\n";
+	output << "\t\t}\n";
+	output << "\t\tallow = { ai = no }\n";
+	output << "\t\teffect = { clr_country_flag = gtfo set_country_flag = gtfo_disabled }\n";
+	output << "\t}\n";
+
+	for (const auto& [tag, country]: countries)
+	{
+		if (skip.contains(tag))
+			continue;
+		const auto& commonCountryFile = country->getCommonCountryFile();
+		const auto& dotPos = commonCountryFile.find_first_of('.');
+		const auto& countryName = commonCountryFile.substr(0, dotPos);
```

Comment:
No, you can't rely on filename to give you a proper country name. You need to look up locs, and I'm pretty sure we're scraping both vanilla locs and whatever other source we may need. In fact, you already have Localisation localisation in every V2 country, ping that.